### PR TITLE
Move focus to systolic when backspace is pressed on an empty diastolic

### PR DESCRIPTION
**Pull request by saket**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/bp/entry/EditTextWithBackspaceListener.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    override fun deleteSurroundingText(beforeLength: Int, afterLength: Int): Boolean {
+      if (beforeLength == 1 && afterLength == 0) {
+        subject.onNext(Any())
+      }
+      return super.deleteSurroundingText(beforeLength, afterLength)
+    }
+  }
+
+  override fun getText(): Editable {
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/bp/entry/EditTextWithBackspaceListener.kt]: Just for my information, why do we need to override this method?

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetController.kt [hunk 2]
```diff
@@ -30,24 +31,49 @@ class BloodPressureEntrySheetController @Inject constructor(
...
+        .map { { ui: Ui -> ui.changeFocusToSystolic() } }
+
+    val deleteLastDigitOfSystolic = diastolicBackspaceClicksWithEmptyText
+        .filter { bp -> bp.systolic.isNotBlank() }
+        .map { bp -> bp.systolic.substring(0, bp.systolic.length - 1) }
+        .map { truncatedSystolic -> { ui: Ui -> ui.setSystolic(truncatedSystolic) } }
+
+    return moveFocusToDiastolic
+        .mergeWith(moveFocusBackToSystolic)
+        .mergeWith(deleteLastDigitOfSystolic)
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetController.kt]: No specific reasons. Just like that.

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt [hunk 3]
```diff
@@ -16,3 +15,5 @@ class BloodPressureDiastolicTextChanged(val diastolic: String) : UiEvent {
 class BloodPressureSaveClicked : UiEvent {
   override val analyticsName = "Blood Pressure Entry:Save Clicked"
 }
+
+class BloodPressureDiastolicBackspaceClicked : UiEvent
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: You're right. I'll fix this.

File: app/src/main/java/org/simple/clinic/bp/entry/EditTextWithBackspaceListener.kt (hunk 1 above)
Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/EditTextWithBackspaceListener.kt]: Discussed in person: `AppCompatEditText.getText()` returns a nullable value if the text is not editable, which will not be the case for an `EditText` that can listen to backpresses.

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: Done

**Pull request merged**